### ISSUE TITLE
Error reporting problem: 'Instant timeouts' in thin-auth

### ISSUE THREAD

**Issue by noahgibbs** [skotostech/thin-auth#3]
I'm hitting a bug which you (@shannona) might recognise. I'm filing this in case you do.

Several views, including verify-email.php, check that the user is logged in and redirect to login.php if they're not. Unfortunately, they don't save $complaint when that happens - any error when trying to verify the login displays as if it's a timeout when you get to login.php.

For instance, here's the relevant code starting from line 17 of verify-email.php:

```
  if ($user == "" || checkTimeOut($auth_sock, $user, $pass, $complaint)) {
     Header("Location: login.php?email-code=$email_code&timeout=1&timeout_from=" .
	    urlencode($_SERVER['PHP_SELF'] . ($_SERVER['QUERY_STRING'] ? "?" . $_SERVER['QUERY_STRING'] : "")));
     exit;
  }
```

As you see, $complaint isn't kept and the problem will be shown as a timeout even if it's a different error.

Do you know offhand what I'm probably doing wrong? Otherwise I'll just drill down to find my problem and improve the errors shown a bit and see if I can easily write $complaint to the logfile.

Thanks!

**Comment by shannona** [comment]
off-hand, no. There was a lot of old code that then got incorporated into a new code base, so it could be a legacy problem or a logic problem. 

I'd think the complaint should either be stored as a cookie or passed forward on that header as GET.

**Comment by noahgibbs** [comment]
Thanks! I'll see if I can find it.

**Comment by shannona** [comment]
Sure thing. Just keep in mind that thin-auth was reverse engineered, not designed. It was built to accommodate the messages built into SkotOS, with maybe 10-20% of the PHP code inherited from the old web portals for the previous iteration of the database. And, it was built on a very tight timeframe.

So, if you're running into a problem where things aren't operating as expected, it's probably not you.

**Comment by noahgibbs** [comment]
Makes sense. It definitely had kinda that vibe :-)

**Comment by noahgibbs** [comment]
I'm going to close this. It looks like the "instant timeout" was related to setting an incorrect subdomain for cookies - didn't realise the gameURL needed to be just a domain name. I've also added a PR with some debug info, only sort of related.

There's a great diagnostic already where the cookie name "biscuit" is used to see if cookies can be set at all. Not sure why I didn't see that the first time I tried this and it failed, but I did the second and fixed it.
